DeepFM-copy-2.py

```
@@ -0,0 +1,385 @@
+import numpy as np
+import tensorflow as tf
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.metrics import roc_auc_score
+from time import time
+from tensorflow.contrib.layers.python.layers import batch_norm as batch_norm
+from yellowfin import YFOptimizer
+
+
+class DeepFM(BaseEstimator, TransformerMixin):
+    def __init__(self, feature_size, field_size,
+                 embedding_size=8, dropout_fm=[1.0, 1.0],
```

Ruff (0.8.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)